content/articles/a-look-at-c#-coding-conventions/index.md

```diff
@@ -0,0 +1,238 @@
+<!--more-->
+The article's aim is to serve as a primary source of information and efficiency for programmers.
+
+Benefits of coding conventions:
+1. They make copying, changing, and maintaining code a lot easier.
+2. They help readers understand the code faster by allowing them to make assumptions based on prior knowledge.
+3. They give the code a unified design so readers may concentrate on the content rather than the layout.
+4. They tell the best way to use the C# programming language in the most ideal manner.
+
+Several coding conventions are used in the C# programming language:
+1. Naming conventions
+2. Layout conventions
+3. Commenting conventions
+
+There are a few naming conventions that should be followed while writing C# code. All information about public elements applies as well to protected and protected internal elements in the following scenarios, which are all supposed to be visible to external callers.
+
+### Pascal's case
+In the Pascal case, every first letter of every word is capitalized as displayed in the example below. One can use the Pascal case for identifiers with three or more characters.
+
+When naming a class, record, or struct, use pascal casing ("PascalCasing").
+
+```
+PublicSchool1
+```
+
+```c#
+public class PascalCase1
+{
+}
+public record PublicIntitution1(
+    string Country,
+    string County,
+    string Village,
+    );
+public struct ValueCoordinate1
+{
+}
+```
+
+Besides to prefixing the name with an I, when naming an interface, use the pascal casing. Customers can tell it's a user interface because of this.
+
+```c#
+public interface IPublicInstitution
+{
+}
+```
+
+Besides to what it's already there, public members of types can be named using the pascal case.
+
+```c#
+public class PracticeExample1
+{
+    // public field
+    public bool IsValid;
+
+    // An init-only property is a property that allows you to do something only once.
+    public IBuilding Building { get; init; }
+
+    // An example of an event
+    public event Action EventProcessing;
+
+    // Implementation of a method
+    public void StartEventProcessing()
+    {
+        // Use of a local function
+        static int CountQueueItems() => WorkerQueue.Count;
+
+    }
+}
+```
+
+Since parameters are public properties of the record, a reader can use pascal casing when putting down positional accounts.
+
+```c#
+public record PublicIntitution
+(
+    string Country,
+    string County,
+    string Village,
+);
+```
+
+### Camel case
+The camelCasing convention, which is used for parameter names, capitalizes the first letter of each word except the first.
+
+```
+voidMain
+```
+
+Another implementation of the camel case is when naming private or internal fields.
+
+```
+camelCasing
+```
```

Comment:
Is this supposed to be here?

Reply:
I was giving it out as an example but I have realized that I positioned it at the wrong place. I opt to remove it